torch/csrc/autograd/python_variable.cpp

```diff
@@ -771,6 +773,79 @@ static PyObject* THPVariable_make_wrapper_subclass(
   END_HANDLE_TH_ERRORS
 }
 
+// DTensor-specific variant of make_wrapper_subclass to minimize DTensor
+// overhead.
+static PyObject* THPVariable_make_dtensor(
```

Comment:
What's the difference with the original one? Is this whole thing to avoid one extra pybind round trip? That should be a few 100ns at best?

Reply:
it is several pybind roundtrips -- each time the Python code messes with a DispatchKey or DispatchKeySet, that's another one.

Reply:
creating a specialized _make_dtensor also allows us to avoid several (pybind!) property accesses on local_tensor that would otherwise be needed to call the generic _make_wrapper_subclass.